src/components/MapView/Analyser/index.tsx

```diff
@@ -153,40 +244,24 @@ function Analyser({ classes }: AnalyserProps) {
   ));
 
   const clearAnalysis = () => dispatch(clearAnalysisResult());
-
-  const runAnalyser = async () => {
-    if (!adminBoundariesExtent) {
-      return;
-    } // hasn't been calculated yet
-
-    if (!selectedDate) {
-      throw new Error('Date must be given to run analysis');
-    }
-
-    if (!hazardLayerId || !baselineLayerId) {
-      throw new Error('Layer should be selected to run analysis');
-    }
-
-    const selectedHazardLayer = LayerDefinitions[
-      hazardLayerId
-    ] as WMSLayerProps;
-    const selectedBaselineLayer = LayerDefinitions[
-      baselineLayerId
-    ] as NSOLayerProps;
-
-    const params: AnalysisDispatchParams = {
-      hazardLayer: selectedHazardLayer,
-      baselineLayer: selectedBaselineLayer,
-      date: selectedDate,
-      statistic,
-      extent: adminBoundariesExtent,
-      threshold: {
-        above: parseFloat(aboveThreshold) || undefined,
-        below: parseFloat(belowThreshold) || undefined,
-      },
-    };
-
-    await dispatch(requestAndStoreAnalysis(params));
+  const shareAnalysis = () => {
+    navigator.clipboard.writeText(
```

Comment:
Remember when I said in a previous comment we should make all the fields live in *one* state object? This is because now, we can construct this url in a scalable way, since we'll have access to the value *and* the key.
```ts
Object.entries(formFields).reduce((search, [key, value])=>search+=key+'='+value+'&','?');
```
This is very rough (doesn't stop adding the `&` at the end but hoping you get the idea ;)